src/analysis/inner_product_space/lax_milgram.lean

```diff
@@ -92,16 +92,16 @@ begin
   rw ← B♯.range.orthogonal_orthogonal,
   rw submodule.eq_top_iff',
   intros v w mem_w_orthogonal,
-  rcases coercive with ⟨C, C_ge_0, coercivity⟩,
-  have : C * ∥w∥ * ∥w∥ ≤ 0 :=
-  calc C * ∥w∥ * ∥w∥
-       ≤ B w w : coercivity w
-  ...  = ⟪B♯ w, w⟫_ℝ : by simp
-  ...  = 0 : mem_w_orthogonal _ ⟨w, rfl⟩,
-  have : ∥w∥ * ∥w∥ ≤ 0 := by nlinarith,
-  have h : ∥w∥ = 0 := by nlinarith [norm_nonneg w],
-  have w_eq_zero : w = 0 := by simpa using h,
-  simp [w_eq_zero],
+  rcases coercive with ⟨C, C_pos, coercivity⟩,
+  obtain rfl : w = 0,
+  { rw [←norm_eq_zero, ←mul_self_eq_zero, ←mul_right_inj' C_pos.ne', mul_zero, ←mul_assoc],
+    apply le_antisymm,
+    { calc C * ∥w∥ * ∥w∥
+          ≤ B w w : coercivity w
+      ... = ⟪B♯ w, w⟫_ℝ : (continuous_linear_map_of_bilin_apply ℝ B w w).symm
+      ... = 0 : mem_w_orthogonal _ ⟨w, rfl⟩ },
+    { exact mul_nonneg (mul_nonneg C_pos.le (norm_nonneg w)) (norm_nonneg w) } },
```

Comment:
As a note, this type of proof becomes a bit easier if working with `nnnorm`, as the nonnegativity assumptions are all trivial to discharge.